File: 0014-longest-common-prefix/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int len=strs.size();
        int idx=strs[0].size();
        int sz0=idx;
        for(int i=1;i<len;i++){
            int sz=strs[i].size();
            int tmp=0;
            for(int j=0;j<min(sz,sz0);j++){
                if(strs[0][j]!=strs[i][j])break;
                tmp++;
                }
            idx=min(idx,tmp);
        }
        string s=strs[0].substr(0,idx);
        return s;
    }
};

File: 0014-longest-common-prefix/solution.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        int mx_len = strs[0].Length;
        char[] tmp = new char[mx_len];
        for(int i=0;i<mx_len;i++){
            tmp[i] = strs[0][i];
        }
        int r = mx_len;
        int len = strs.Length;
        for(int i=1;i<len; i++){
            int now_len = Math.Min(mx_len,strs[i].Length);
            mx_len = 0;
            for(int j=0;j<now_len;j++){
                if(tmp[j]==strs[i][j])mx_len=j+1;
                else break;
            }
        }
        string ans = "";
        if(mx_len>0)ans = new string(tmp, 0, mx_len);
        return ans;
    }
}
